### ISSUE TITLE
spatial/r3: Add method to Rotation for stringing multiple rotations

### ISSUE THREAD

**Issue by soypat** [gonum/gonum#1804]
Add a way to combine rotations to avoid this: `r := Rotation(quat.Mul(quat.Number(q), quat.Number(r)))`
```go
// Then combines rotations r and q such that 
// q follows r.
func (r Rotation) Then(q Rotation) Rotation {
	return Rotation(quat.Mul(quat.Number(q), quat.Number(r)))
}
```

**Comment by kortschak** [comment]
I would prefer not to add layers of abstraction here.

**Comment by soypat** [comment]
I agree the function is sufficiently short to be written by users. My main gripe was that I spent quite a while researching quaternion rotation composition. It wasn't until I came across a stackoverflow post with the conjugate equation for two quaternion rotations. It'd be nice if users have a way of combining two rotations that the library guarantees works since it saves the less-traveled programmers of us some research and worries about correctness.

**Comment by sbinet** [comment]
alternatively: what about an example?

**Comment by sbinet** [comment]
FTR, I was hinting at a dedicated example, say:
```go
func ExampleRotation_compose() { ... }
```
(so it's displayed together with the `Rotation` type.)

but, FTR, I am fine with adding a dedicated constructor.

**Comment by kortschak** [comment]
I think an example is an excellent thing to add for this. It's worth pointing out that there is a way to compose rotations, `Mul`. It may be that there is a need for a higher level  abstraction on quat, but it doesn't need to be here.
